metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -37,6 +37,14 @@ CanonicalDBG::CanonicalDBG(std::shared_ptr<const DeBruijnGraph> graph, size_t ca
         if (graph_->alphabet()[i] == boss::BOSS::kSentinel)
             has_sentinel_ = true;
     }
+
+    if (NodeRC *node_rc = graph_->get_extension_threadsafe<NodeRC>()) {
+        add_extension(std::shared_ptr<NodeRC>(std::shared_ptr<NodeRC>{}, node_rc));
+    } else {
+        auto ext = std::make_shared<NodeRC>();
+        ext->set_graph(*graph_);
+        add_extension(ext);
```

Comment:
At what point does it initialize the index?

Reply:
`NodeRC`'s main purpose is to support the `call_outgoing_from_rc` and `call_incoming_from_rc` methods, so having an index loaded is not a strict requirement for it to function. If no index is available, it simply uses its stored graph pointer to compute those functions.

Reply:
So is it uninitialized then? Why do you add it?

Reply:
The point is that I wanted all of the "call adjacent nodes to the reverse complements" code to be in the same place, so I moved it out of CanonicalDBG and into NodeRC.

So, NodeRC can perform the same task for all kinds of DeBruijnGraphs, even if they don't have an index computed and if they are not DBGSuccinct.

Reply:
Ah, I see. This wasn't obvious from the interface.